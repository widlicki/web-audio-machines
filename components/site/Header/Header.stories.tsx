import { ComponentMeta, ComponentStory } from '@storybook/react'
import Header from './Header'

export default {
  title: 'components/site/Header',
  component: Header,
  parameters: {
    layout: 'fullscreen',
  },
} as ComponentMeta<typeof Header>

const Template: ComponentStory<typeof Header> = (args) => <Header {...args} />

export const LoggedIn = Template.bind({})
LoggedIn.args = {
  isAuthenticated: true,
}

export const LoggedOut = Template.bind({})
LoggedOut.args = {
  isAuthenticated: false,
}
