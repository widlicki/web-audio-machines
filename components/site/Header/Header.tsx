import { AppBar, Link, Toolbar } from '@mui/material'
import Box from '@mui/material/Box'
import theme from '../../../styles/theme'
import NavLink from '../../base/NavLink/NavLink'

export interface HeaderProps {
  isAuthenticated: boolean
}

const Header = ({ isAuthenticated }: HeaderProps) => {
  return (
    <AppBar
      elevation={0}
      position="fixed"
      sx={{
        background: theme.background.color.light,
        borderBottom: `${theme.border.size.md} solid ${theme.palette.secondary.main}`,
        height: theme.spacing(12),
      }}
    >
      <Toolbar
        sx={{ justifyContent: 'space-between', marginTop: theme.spacing(2) }}
      >
        <Box sx={{ flex: 1 }} />
        <Link
          variant="h4"
          href="/"
          sx={{
            fontSize: 24,
            color: 'common.white',
            textDecoration: 'none',
            '&:hover': {
              cursor: 'pointer',
            },
          }}
        >
          web &bull; audio &bull; machines
        </Link>
        <Box
          sx={{
            flex: 1,
            display: 'flex',
            justifyContent: 'flex-end',
          }}
        >
          {!isAuthenticated ? (
            <>
              <NavLink to="/sign-in">Sign In</NavLink>
              <NavLink to="/sign-up">Sign Up</NavLink>
            </>
          ) : (
            <NavLink to="/sign-in">Sign Out</NavLink>
          )}
        </Box>
      </Toolbar>
    </AppBar>
  )
}

export default Header
