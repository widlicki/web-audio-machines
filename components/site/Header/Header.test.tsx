import { render } from '@testing-library/react'
import Header, { HeaderProps } from './Header'
import { LoggedIn, LoggedOut } from './Header.stories'

const defaultArgs: HeaderProps = {
  isAuthenticated: false,
} as HeaderProps

describe('Header Stories', () => {
  new Map([
    ['correct header when user is logged in', LoggedIn],
    ['correct header when user is not logged in', LoggedOut],
  ]).forEach(({ args }, key) => {
    it(`renders ${key}`, () => {
      const component = <Header {...defaultArgs} {...args} />
      const { container } = render(component)
      expect(container).toMatchSnapshot()
    })
  })
})
