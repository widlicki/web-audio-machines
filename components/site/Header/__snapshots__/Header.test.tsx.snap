// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Stories renders correct header when user is logged in 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-xnb1ep-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1wlywpi-MuiToolbar-root"
    >
      <div
        class="MuiBox-root css-1rr4qq7"
      />
      <a
        class="MuiTypography-root MuiTypography-h4 MuiLink-root MuiLink-underlineAlways css-l2y2e4-MuiTypography-root-MuiLink-root"
        href="/"
      >
        web • audio • machines
      </a>
      <div
        class="MuiBox-root css-1j9bsaa"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-kq33hk-MuiTypography-root-MuiLink-root"
        >
          Sign Out
        </a>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header Stories renders correct header when user is not logged in 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-xnb1ep-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1wlywpi-MuiToolbar-root"
    >
      <div
        class="MuiBox-root css-1rr4qq7"
      />
      <a
        class="MuiTypography-root MuiTypography-h4 MuiLink-root MuiLink-underlineAlways css-l2y2e4-MuiTypography-root-MuiLink-root"
        href="/"
      >
        web • audio • machines
      </a>
      <div
        class="MuiBox-root css-1j9bsaa"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-kq33hk-MuiTypography-root-MuiLink-root"
        >
          Sign In
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-kq33hk-MuiTypography-root-MuiLink-root"
        >
          Sign Up
        </a>
      </div>
    </div>
  </header>
</div>
`;
